index.php

```diff
@@ -1,477 +1,102 @@
-            }
-
-            // Email
-            if(strlen($email) < 5) {
-                throw new RuntimeException("Please enter a valid email address!");
-            }
-
-            /** @noinspection RegExpRedundantEscape */
-            $email_pattern = '/^[\w\-\.\+]+@[\w\-\.]+\.[a-zA-Z]{2,4}$/';
-            if(!preg_match($email_pattern, $email)) {
-                throw new RuntimeException("Please enter a valid email address!");
-            }
-
-            // Check for hotmail
-
-            $email_arr = explode('@', $email);
-            $email_arr[1] = strtolower($email_arr[1]);
-
-            if(array_search($email_arr[1], System::UNSERVICEABLE_EMAIL_DOMAINS) !== false) {
-                throw new RuntimeException(implode(' / ', System::UNSERVICEABLE_EMAIL_DOMAINS) . " emails are currently not supported!");
-            }
-
-            // Check for username/email existing
-            $result = $system->db->query(
-                "SELECT `user_id`, `user_name`, `email` FROM `users`
-                    WHERE `email`='$email' OR `user_name`='$user_name' LIMIT 1"
-            );
-            if(mysqli_num_rows($result) > 0) {
-                $result = mysqli_fetch_assoc($result);
-                if(strtolower($result['user_name']) == strtolower($user_name)) {
-                    throw new RuntimeException("Username already in use!");
-                }
-                else if(strtolower($result['email']) == strtolower($email)) {
-                    throw new RuntimeException("Email address already in use!");
-                }
-            }
-
-            // Gender
-            if(!in_array($gender, User::$genders, true)) {
-                throw new RuntimeException("Invalid gender!");
-            }
-
-            // Village
-            // Load villages
-            $result = $system->db->query("SELECT `villages`.`name`, `x`, `y`, `map_id` FROM `villages`
-                INNER JOIN `maps_locations` on `villages`.`map_location_id` = `maps_locations`.`location_id`
-            ");
-            $villages = [];
-            while ($row = mysqli_fetch_array($result)) {
-                $villages[$row['name']] = $row;
-            }
-            if(!isset($villages[$village])) {
-                throw new RuntimeException("Invalid village!");
-            }
-
-            // Encrypt password
-            $password = $system->hash_password($password);
-
-            $verification_code = sha1(mt_rand(1, 1337000));
-
-            $village_coords = new TravelCoords($villages[$village]['x'], $villages[$village]['y'], $villages[$village]['map_id']);
+    $route = Router::$routes['home'];
+    $system->layout->renderBeforeContentHTML(
+        system: $system,
+        player: null, // No need to check here, no session set
+        page_title: $route->title,
+        render_header: $route->render_header, render_sidebar: $route->render_sidebar,
+        render_topbar: $route->render_topbar, render_content: $route->render_content
```

Comment:
Would `$system->layout->renderStaticPageHeader($route->title)` work for you here?

We can do this cleanup after, but I think that or something is a preferable option over having all these parameters that disable a lot of what this function does, for only this case

Reply:
![image](https://github.com/elementum-games/shinobi-chronicles/assets/24821372/5fc3d330-7e62-458a-94ad-f878c59ebb88)
I think this is a solid plan, just need to do some tweaks with Layout to make it work later.